src/ngraph/runtime/cpu/cpu_backend.cpp

```diff
@@ -112,7 +110,7 @@ bool runtime::cpu::CPU_Backend::call(shared_ptr<Function> func,
     FunctionInstance& instance = m_function_map[func];
     if (instance.m_external_function == nullptr)
     {
-        rc = compile(func);
+        throw runtime_error("compile() must be called before call().");
```

Comment:
is there any reason `call(...)` cannot internally call `compile(...)`? But a better option is to change the signature - as I proposed above.

Reply:
Yes, because call will ultimately take a Handle, not a shared_ptr<Function> so there will be no way it can compile. If you load a saved compiled model you will not get a shared_ptr<Function> when you load it, you will get a handle.  This PR is just one of several.

Reply:
Ok - that makes sense.